src/world.rs

```diff
@@ -912,6 +912,31 @@ impl World {
     pub fn is_empty(&self) -> bool {
         self.len() == 0
     }
+
+    /// Entity handles that will be yielded by future spawns
+    ///
+    /// [`Entity`] handles will be allocated deterministically between different
+    /// worlds (e.g. across serialization) if they have the same live entities
+    /// and the same freelist.
+    ///
+    /// To deserialize a world that will match the entity allocation behavior of
+    /// a previously serialized world, first create a new world, then use
+    /// [`spawn_at`](Self::spawn_at) or similar to reconstruct the serialized
+    /// world's entities in place, then call
+    /// [`set_freelist`](Self::set_freelist) with the serialized world's
+    /// freelist.
+    pub fn freelist(&self) -> impl ExactSizeIterator<Item = Entity> + '_ {
+        self.entities.freelist()
+    }
+
+    /// Designate entity IDs to be reused by future spawns
+    ///
+    /// See [`freelist`](Self::freelist). May only be called when the set of
+    /// live entity IDs exactly matches those that were live when the supplied
+    /// freelist was captured.
```

Comment:
I was thinking about how Entities clears its `pending` vec without calling `flush` and whether that's ok in weird cases like `deserialize, spawn, despawn, reserve entity, set freelist, use reserved entity ID` - and I think it is because this second sentence here is supposed to rule out any case where the `pending` vec matters.

But it might be worth spelling it out with another docs sentence? Something like `For example, spawning, despawning, or reserving an entity ID would invalidate a previously captured freelist.`